cmd/zetatool/inbound/bitcoin.go

```diff
@@ -0,0 +1,233 @@
+package inbound
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	cosmosmath "cosmossdk.io/math"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/node/cmd/zetatool/config"
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/pkg/coin"
+	"github.com/zeta-chain/node/pkg/rpc"
+	crosschaintypes "github.com/zeta-chain/node/x/crosschain/types"
+	"github.com/zeta-chain/node/x/observer/types"
+	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/common"
+	zetaclientObserver "github.com/zeta-chain/node/zetaclient/chains/bitcoin/observer"
+)
+
+func btcInboundBallotIdentifier(
+	cfg config.Config,
+	zetacoreClient rpc.Clients,
+	inboundHash string,
+	inboundChain chains.Chain,
+	zetaChainID int64) (string, error) {
+	params, err := chains.BitcoinNetParamsFromChainID(inboundChain.ChainId)
+	if err != nil {
+		return "", fmt.Errorf("unable to get bitcoin net params from chain id: %s", err)
+	}
+
+	connCfg := &rpcclient.ConnConfig{
+		Host:         cfg.BtcHost,
+		User:         cfg.BtcUser,
+		Pass:         cfg.BtcPassword,
+		HTTPPostMode: true,
+		DisableTLS:   true,
+		Params:       params.Name,
+	}
+	rpcClient, err := rpcclient.New(connCfg, nil)
+	if err != nil {
+		return "", fmt.Errorf("error creating rpc client: %s", err)
+	}
```

Comment:
Please use the new internal bitcoin rpc client introduced in https://github.com/zeta-chain/node/pull/3349 if possible